### ISSUE TITLE
In JSX components, spreading props breaks component's syntax highlighting.

### ISSUE THREAD

**Issue by mithi** [PrismJS/prism#2753]
**Information**
- Language: [JSX]
- Plugins: I'm using prism through [React Markdown](https://github.com/remarkjs/react-markdown) and [React Syntax Highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter)

**Description**

In JSX components, spreading props breaks component's syntax highlighting. 

**Code snippet**

[Test page](https://prismjs.com/test.html#language=jsx&text=const%20Test%20%3D%20()%20%3D%3E%20%7B%0A%20%20%20%20return%20(%0A%20%20%20%20%20%20%20%3Cdiv%3E%0A%20%20%20%20%20%20%20%20%20%20%20%3CButton%20%7B...%7BonClick%2C%20disabled%7D%7D%3E%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20Click%20(Wrong%20Highlighting)%0A%20%20%20%20%20%20%20%20%20%20%20%3C%2FButton%3E%0A%20%20%20%20%20%20%20%20%20%20%20%3CButton%20onClick%3D%7BonClick%7D%20disabled%3D%7Bdisabled%7D%3E%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20Click%20(Correct%20highlighting)%0A%20%20%20%20%20%20%20%20%20%20%20%3C%2FButton%3E%0A%20%20%20%20%20%20%20%3C%2Fdiv%3E%0A%20%20%20%20)%0A%7D)

Here's Github's version

```jsx
const Test = () => {
    return (
       <div>
           <Button {...{onClick, disabled}}>
               Click (Wrong Highlighting)
           </Button>
           <Button onClick={onClick} disabled={disabled}>
               Click (Correct highlighting)
           </Button>
       </div>
    )
}
```

**Screenshots** (images)

Prism's broken syntax highlighting, breaks the colors of the first component

<img width="739" alt="Screen Shot 2021-02-07 at 7 44 59 PM" src="https://user-images.githubusercontent.com/1670421/107145579-94033a00-697d-11eb-8005-5c92c9592a12.png">

Github correctly highlights it, the first component is highlighted correctly

<img width="751" alt="Screen Shot 2021-02-07 at 7 46 05 PM" src="https://user-images.githubusercontent.com/1670421/107145589-a5e4dd00-697d-11eb-8a94-c04a49d638a1.png">

I have seen the issues regarding `Nested objects inside props` and I'm thinking they might be related. 
https://github.com/PrismJS/prism/issues/2598
https://github.com/PrismJS/prism/issues/1548

Is this expected / known behavior, and is this going to be fixed at all? Thanks!

**Comment by RunDevelopment** [comment]
That's definitely a bug on our side. The current implementation only allows for variables in spread attributes (e.g. `{...foo}`) but nothing else. This is fixed in #2754 and will be available soon.

**Comment by mithi** [comment]
> That's definitely a bug on our side. The current implementation only allows for variables in spread attributes (e.g. `{...foo}`) but nothing else. This is fixed in #2754 and will be available soon.

This is awesome! Btw, if it's okay to ask, are you planning to release `1.24.0` soon? I see here that `1.23.0` was released a month ago, and you've got 29 commits since then. Just curious. Thanks!

**Comment by RunDevelopment** [comment]
The next release will be coming soon (maybe in a week or so). In the meantime, you can always get the latest version of Prism from our [download page](https://prismjs.com/download.html).
